types/mydecimal.go

```diff
@@ -2345,3 +2345,21 @@ func NewMaxOrMinDec(negative bool, prec, frac int) *MyDecimal {
 	terror.Log(errors.Trace(err))
 	return dec
 }
+
+// NewZeroDec returns the zero value decimal for given precision and fraction.
+func NewZeroDec(prec, frac int) *MyDecimal {
+	if prec == UnspecifiedLength {
+		return NewDecFromInt(0)
+	}
+	str := make([]byte, prec+1)
+	for i := 0; i < len(str); i++ {
+		str[i] = '0'
+	}
+	if frac != UnspecifiedLength {
+		str[prec-frac] = '.'
+	}
+	dec := new(MyDecimal)
+	err := dec.FromString(str)
+	terror.Log(errors.Trace(err))
```

Comment:
Please return this error instead of just printing it.

Reply:
Function `NewMaxOrMinDec` in this file is similar to this (just print the error). Also, functions that call `NewZeroDec` does not return error (`AllocPartialResult`). So I think it's better not to return an error.